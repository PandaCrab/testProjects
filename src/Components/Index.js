import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
