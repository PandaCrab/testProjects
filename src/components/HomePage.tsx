import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
