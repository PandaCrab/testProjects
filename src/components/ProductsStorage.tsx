import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
