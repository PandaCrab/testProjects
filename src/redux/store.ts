import { compose, createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import thunk from 'redux-thunk';

import stuffReducer, { stuffWatcher } from './ducks/stuff';
import dataReducer, { dataWatcher } from './ducks/data';
import addressReducer, { 
  addressInputWatcher,
  addressWatcher,
  navigatorAddressWatcher,
  geolocationWatcher
} from './ducks/address';

const saga = createSagaMiddleware();

const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] as typeof compose || compose;
 
export const store = createStore(combineReducers({
    order: stuffReducer,
    data: dataReducer,
    address: addressReducer
  }), composeEnhancers(
    applyMiddleware(
      thunk, saga
    )
  ));

saga.run(stuffWatcher);
saga.run(dataWatcher);
saga.run(addressWatcher);
saga.run(addressInputWatcher);
saga.run(navigatorAddressWatcher);
saga.run(geolocationWatcher);
