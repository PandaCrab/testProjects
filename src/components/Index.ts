import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
