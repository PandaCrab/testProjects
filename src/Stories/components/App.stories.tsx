import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
