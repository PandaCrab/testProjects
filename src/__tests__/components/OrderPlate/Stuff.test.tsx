import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { Stuff } from '../../../components/Index';
import { mount } from 'enzyme';
import { Store, AnyAction } from 'redux';

describe('Stuff element', () => {
    const initialState = {order: {stuff: [
            {
                'id': 1,
                'imgUrl': `https://guesseu.scene7.com/is/image/GuessEU/
                    M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80
                    &op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0`,
                'name': 'Check Print Shirt',
                'color': 'Grey+red+black',
                'price': 110
            }
        ],
        loading: false
    }}
    const mockStore = configureStore();
    let store: Store<Object, AnyAction> = mockStore(initialState);

    const component = mount(
        <Provider store={store}>
            <Stuff />
        </Provider>
    );

    it('should render correct', () => {
        expect(component).toMatchSnapshot();
    });

    describe('render whithout stuff', () => {
        beforeAll(() => store = mockStore({order: {stuff: [], loading: false}}));
        afterAll(() => store = mockStore(initialState));

        it('should rendered without stuff', () => {
            expect(component).toBeTruthy();    
        });
    });

    describe('render loader', () => {
        beforeAll(() => store = mockStore({order: {stuff: [], loading: false}}));
        afterAll(() => store = mockStore(initialState));
        it('should render loader when stuff empty', () => {
            expect(component.find('LoaderContainer')).toBeTruthy();
        });
    });

    describe('render loader container when server shut down', () => {
        beforeAll(() => store = mockStore({order: {stuff: [], loading: false}}));
        afterAll(() => store = mockStore(initialState));

        it('should render loader when stuff is null', () => {
            expect(component.find('LoaderContainer')).toBeTruthy();
        });
    });
});
