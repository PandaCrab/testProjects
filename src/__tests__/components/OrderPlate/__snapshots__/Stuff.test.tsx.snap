// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stuff element should render correct 1`] = `
<section
  className="sc-hHLeRK dPNuhk"
>
  <img
    alt="Check Print Shirt"
    className="sc-dmRaPn zCVrD"
    src="https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0"
  />
  <section
    className="sc-kgflAQ dbCiuf"
  >
    <h6
      className="sc-fLlhyt clxgpt"
    >
      Check Print Shirt
    </h6>
    <p
      className="sc-bBrHrO bXHGEt"
    >
      Grey+red+black
    </p>
    <p
      className="sc-bBrHrO bXHGEt"
    >
      Quantity: 1
    </p>
  </section>
  <p
    className="sc-ivTmOn bpCsUh"
  >
    110
    $
  </p>
</section>
`;
